contracts/multiswap-contracts/FiberRouter.sol

```diff
@@ -619,444 +514,144 @@ function _swapOnSameNetwork(
+        require(amountOut >= minAmountOut, "FR: Slippage check failed");
+        return amountOut;
     }
 
-    /**
-     * @dev Handles the execution of the `fillOrderTo` operation, involving 1inch aggregator
-     * @param to The recipient address to receive the swapped tokens
-     * @param fromToken The address of the input token for the swap (foundryToken or takerAsset)
-     * @param amountIn The amount of input tokens to be swapped
-     * @param oneInchData The data containing information for the 1inch swap
-     * @return returnAmount The amount of tokens received after the swap and transaction execution
-     */
-    function handleFillOrderTo(
-        address payable to,
-        address fromToken,  // foundryToken // takerAsset
-        uint256 amountIn,
-        bytes memory oneInchData
-    ) internal returns (uint256 returnAmount) {
-        // Decoding oneInchData to get the required parameters
-        (
-            Order memory order_,
-            bytes memory signature,
-            bytes memory interaction,
-            uint256 makingAmount,
-            uint256 takingAmount,  // destinationAmountIn
-            uint256 skipPermitAndThresholdAmount,
-            address target  // receiverAddress
-        ) = abi.decode(
-            oneInchData,
-            (Order, bytes, bytes, uint256, uint256,uint256, address)
-        );
-
-        // Manually create a new Order for IOneInchSwap
-        IOneInchSwap.Order memory oneInchOrder = IOneInchSwap
-            .Order({
-                salt: order_.salt,
-                makerAsset: order_.makerAsset,
-                takerAsset: order_.takerAsset,
-                maker: order_.maker,
-                receiver: order_.receiver,
-                allowedSender: order_.allowedSender,
-                makingAmount: order_.makingAmount,
-                takingAmount: order_.takingAmount,
-                offsets: order_.offsets,
-                interactions: order_.interactions
-            });
-
-        // Perform additional checks and validations if needed
-        require(to == target, "FR: recipient address bad oneInch Data");
-        require(fromToken == order_.takerAsset, "FR: takerAsset bad oneInch Data");
-        require(amountIn == takingAmount, "FR: inputAmount bad oneInch Data ");
-        require(oneInchData.length >= 4, "Data too short for valid call");
-
-        // Performing the swap
-        ( returnAmount, , ) = IOneInchSwap(oneInchAggregatorRouter).fillOrderTo(
-            oneInchOrder,
-            signature,
-            interaction,
-            makingAmount,
-            takingAmount,
-            skipPermitAndThresholdAmount,
-            target
-        );
-
-        emit SwapHandled(
-            oneInchAggregatorRouter,
-            to,
-            fromToken,
-            amountIn,
-            returnAmount // should be returned 
-        );
+    function _getBalance(address token, address account) public view returns (uint256) {
```

Comment:
Why a public function with name starting with `_`. Should this be private?

Reply:
woops my bad, that's correct